Package/Image/ImageResolver.cs

```diff
@@ -37,6 +37,30 @@ public ImageResolution ResolveImage(ImageSpecifier image, PackageDependencyGraph
                     var pkg2 = packages[j];
                     if (pkg2.Name == pkg1.Name)
                     {
+                        // Trivially remove one of the completely identical packages
+                        if (pkg2.Version == pkg1.Version)
+                        {
+                            packages.RemoveAt(j);
+                            goto retry;
+                        }
+                        
+                        // In the case where the versions are the same, but the match behavior is different
+                        // If one is exact and the other is compatible, we can remove the compatible one.
+                        if (pkg1.Version.CompareTo(pkg2.Version) == 0)
+                        {
+                            if (pkg1.Version.MatchBehavior == VersionMatchBehavior.Exact && pkg2.Version.MatchBehavior == VersionMatchBehavior.Compatible)
+                            {
+                                packages.RemoveAt(j);
+                                goto retry;
+                            }
+                        
+                            if (pkg2.Version.MatchBehavior == VersionMatchBehavior.Exact && pkg1.Version.MatchBehavior == VersionMatchBehavior.Compatible)
+                            {
+                                packages.RemoveAt(i);
+                                goto retry;
```

Comment:
This looks like a bug. 

`pkg1` is assigned right before the `retry` label, and this goto will continue the inner loop without reassigning `pkg1`.

This should probably be:

```
packages[i] = packages[j];
packages.RemoveAt(j);
goto retry;
```